plugins/bnet_plugin/bnet_plugin.cpp

```diff
@@ -117,16 +119,16 @@ using std::vector;
 struct hello {
    public_key_type               peer_id;
    string                        network_version;
-   string                        agent;
-   string                        protocol_version = "1.0.0";
+   string                        meta;
```

Comment:
This string-based extensions sets a bad precedent IMO.  I understand we are trying to rush a feature into an experimental plugin but we should probably address the underlying need for versioning in this protocol now instead of kicking the can down the road.  

I would suggest we deprecate `hello` by renaming it to `legacy_hello` and maintaining its location in the static_variant `hello` structure which is easier to version and support backwards/forwards compat on.

To this end, we should probably put statically sized version (whatever width int we think we need) as the first member so that peeking at a `hello` can tell you enough to know if you support the protocol version without unpacking the data.

We should also have a `vector<extensions>` as the last part of the basic hello, where an `extension` is a static_variant of 1 or more things.  We could move things like `pending_block_ids` and `user/password` into extensions so that the base `hello` is lean.  We can also take this chance to prune stuff that wasn't/isn't used